File: movie-search-backend/src/main/java/com/moviesearch/entity/Movie.java
package com.moviesearch.entity;

import lombok.Data;
import java.time.LocalDateTime;

@Data
public class Movie {
    private String imdbId;
    private String title;
    private String year;
    private String rated;
    private String released;
    private String runtime;
    private String genre;
    private String director;
    private String writer;
    private String actors;
    private String plot;
    private String language;
    private String country;
    private String awards;
    private String poster;
    private String imdbRating;
    private String imdbVotes;
    private String type;
    private String dvd;
    private String boxOffice;
    private String production;
    private String website;
    private LocalDateTime cachedAt;
}

File: movie-search-frontend/src/types/movie.ts
export interface Movie {
  imdbId: string;
  title: string;
  year: string;
  rated: string;
  released: string;
  runtime: string;
  genre: string;
  director: string;
  writer: string;
  actors: string;
  plot: string;
  language: string;
  country: string;
  awards: string;
  poster: string;
  imdbRating: string;
  imdbVotes: string;
  type: string;
  dvd: string;
  boxOffice: string;
  production: string;
  website: string;
  cachedAt: string;
}

export interface MovieSearchRequest {
  search: string;
  page: number;
  year?: number;
  type?: string;
}

export interface MovieSearchResponse {
  movies: Movie[];
  totalResults: number;
  currentPage: number;
  totalPages: number;
  hasNextPage: boolean;
  hasPreviousPage: boolean;
  searchTerm: string;
  responseTimeMs: number;
}

export interface FeatureFlag {
  name: string;
  enabled: boolean;
}

export interface FeatureFlagBatchResponse {
  flags: Record<string, FeatureFlag>;
}

export interface ApiError {
  code: number;
  error: string;
  message: string;
  timestamp: string;
}

export interface ValidationError {
  code: number;
  error: string;
  message: string;
  fieldErrors: Record<string, string>;
  timestamp: string;
}
